packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -94,13 +168,114 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
+    return version == null ? key : Uri.parse(
+      key
+        .toString()
+        .split(extension)
+        .join() + '.' + version.toString() + extension
+    );
   }
+}
 
-  File _getGoldenFile(Uri uri) {
-    return fs.directory(basedir).childFile(fs.file(uri).path);
+/// A [FlutterGoldenFileComparator] for testing golden images with Skia Gold.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// images for framework-related golden tests and process results. Currently
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
+///
+/// See also:
+///
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterGoldensRepositoryFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images through Skia Gold.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  FlutterSkiaGoldFileComparator(Uri basedir) : super(basedir) {
+    _skiaClient = SkiaGoldClient(fs.directory(basedir));
+  }
+
+  /// Constructor used specifically for testing in order to provide mock members.
+  FlutterSkiaGoldFileComparator.test(
+    Uri basedir,
+    FileSystem testFileSystem,
+    Platform testPlatform,
+  ) : super(
+    basedir,
+    fs: testFileSystem,
+    platform: testPlatform,
+  );
+
+
+  SkiaGoldClient _skiaClient;
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    GoldensClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= GoldensClient();
+
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(goldens, defaultComparator);
+    final SkiaGoldClient skiaClient = SkiaGoldClient(baseDirectory);
+    await skiaClient.auth();
+    await skiaClient.imgtestInit();
+    return FlutterSkiaGoldFileComparator(baseDirectory.uri);
+  }
+
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
+
+    final File goldenFile = getGoldenFile(golden);
+    if (!goldenFile.existsSync()) {
+      throw TestFailure('Could not be compared against non-existent file: "$golden"');
+    }
+    return await _skiaClient.imgtestAdd(golden.path, goldenFile);
+  }
+
+  /// Prepends the golden Uri with the library name that encloses the current
+  /// test.
+  Uri _addPrefix(Uri golden) {
+    final String prefix = basedir.pathSegments[basedir.pathSegments.length - 2];
+    return Uri.parse(prefix + '.' + golden.toString());
+  }
+}
+
+class FlutterDummyGoldenFileComparator extends FlutterGoldenFileComparator {
```

Comment:
Maybe Call this FlutterSkippingGoldenFileComparator to indicate that it just skips all comparisons?